services/venice-controller/src/main/java/com/linkedin/venice/ingestion/control/RealTimeTopicSwitcher.java

```diff
@@ -222,7 +222,10 @@ public void transmitVersionSwapMessage(Store store, int previousVersion, int nex
 
     // If there exists an RT, then broadcast the Version Swap message to it, otherwise broadcast it to the VT
     String storeName = store.getName();
-    if (!topicManager.containsTopic(pubSubTopicRepository.getTopic(Version.composeRealTimeTopic(storeName)))) {
+    boolean isNextStoreVersionHybrid = nextStoreVersion.isHybrid();
+    String nextStoreVersionRTTopic = Utils.getRealTimeTopicName(nextStoreVersion);
+    if (!(isNextStoreVersionHybrid
+        && topicManager.containsTopic(pubSubTopicRepository.getTopic(nextStoreVersionRTTopic)))) {
```

Comment:
Looks like we changed this condition a little bit, what could happen if `previousStoreVersion` is hybrid, but `nextStoreVersion` is not? In this case, the old code sends it to the RT topic, but the new code sends to VT topic? Maybe we should double-check with the feature owner to understand better.